### ISSUE TITLE
Issue with spaCy backend

### ISSUE THREAD

**Issue by lepennec** [statsmaths/cleanNLP#3]
Hi,

I wanted to to test the new spaCy backend but seems to suffer from an issue. I have installed spaCy with pip and it seems to be accesible by reticulate. I have also downloaded the language pack with `python -m spacy download en` but it seems that I'm missing something as I obtain the following result.

```
> reticulate::py_run_string("import os; import spacy; print(os.path.dirname(spacy.__file__))")
/usr/lib64/python2.7/site-packages/spacy
> init_spaCy()
> text <- "Dummy text."
> obj <- annotate(text, as_strings = TRUE)
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  TypeError: 'errors' is an invalid keyword argument for this function

Detailed traceback: 
  File "<string>", line 38, in processFiles
```

Have you already seen this behaviour? How can I help you and your superb package?

Erwan

Ps: I'm using Linux with Fedora.

**Comment by taylor-arnold** [comment]
Thanks for your note. I've hunted down this bug and have determined that it is an issue with Python 2 versus Python 3, specifically how their respective `open` functions work by default. I made a change to the Python wrapper recently in order to fix another bug and introduced this one at the same time. Please run this to get the updated version:
```{r}
devtools::install_github("statsmaths/cleanNLP")
```
Let me know if that solves the problem, if it persists, or if you run into any other issues.

**Comment by lepennec** [comment]
I have now a new error... but again this seems related to Python 2 vs Python 3. 
```
ann <- annotate(text, as_strings = TRUE)
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  TypeError: write() argument 1 must be unicode, not str
```
As I understand, the way the string are encoded is different between the two versions, and the use of py_unicode from reticulate may help.

Erwan

**Comment by taylor-arnold** [comment]
Okay, this string/unicode issue really became a problem when I changed the write function from the base version to io.write, but only in Python 2. I thought my unit tests were running both versions of Python, but that was not in fact true. I think (hope) that reloading the package now will work. Please give an update when you get a chance to try it.

**Comment by lepennec** [comment]
It's working now! Thanks for your reactivity and your work.

**Comment by taylor-arnold** [comment]
Excellent, glad that worked. Thanks for the bug report. This will surely help others with the same issue.

Fixed with a7eb612cca5a91b8b0e24c91acbb5ff2be1f6800.
